src/text/text_formatter.rs

```diff
@@ -0,0 +1,401 @@
+        if value.is_nan() {
+            write!(self.output, "nan")?;
+            return Ok(());
+        }
+
+        if value.is_infinite() {
+            if value.is_sign_positive() {
+                write!(self.output, "+inf")?;
+            } else {
+                write!(self.output, "-inf")?;
+            }
+            return Ok(());
+        }
+
+        // The {:e} formatter provided by the Display trait writes floats using scientific
+        // notation. It works for all floating point values except -0.0 (it drops the sign).
+        // See: https://github.com/rust-lang/rust/issues/20596
+        if value == 0.0f64 && value.is_sign_negative() {
+            write!(self.output, "-0e0")?;
+            return Ok(());
+        }
+
+        write!(self.output, "{:e}", value)?;
+        Ok(())
+    }
+
+    pub fn format_decimal(&mut self, value: &Decimal) -> IonResult<()> {
+        write!(self.output, "{}", value)?;
+        Ok(())
+    }
+
+    pub fn format_timestamp(&mut self, value: &Timestamp) -> IonResult<()> {
+        let (offset_minutes, datetime) = if let Some(minutes) = value.offset {
+            // Create a datetime with the appropriate offset that we can use for formatting.
+            let datetime: DateTime<FixedOffset> = value.clone().try_into()?;
+            // Convert the offset to minutes --v
+            (Some(minutes.local_minus_utc() / 60), datetime)
+        } else {
+            // Our timestamp has an unknown offset. Per the spec, this means it makes no
+            // assertions about *where* it was recorded, but its fields are still in UTC.
+            // Create a UTC datetime that we can use for formatting.
+            let datetime: NaiveDateTime = value.clone().try_into()?;
+            let datetime: DateTime<FixedOffset> = FixedOffset::east(0).from_utc_datetime(&datetime);
+            (None, datetime)
+        };
+
+        write!(self.output, "{:0>4}", datetime.year())?;
+        //                     ^-- 0-padded, right aligned, 4-digit year
+        if value.precision == Precision::Year {
+            write!(self.output, "T")?;
+            return Ok(());
+        }
+
+        write!(self.output, "-{:0>2}", datetime.month())?;
+        //                   ^-- delimiting hyphen and 0-padded, right aligned, 2-digit month
+        if value.precision == Precision::Month {
+            write!(self.output, "T")?;
+            return Ok(());
+        }
+
+        write!(self.output, "-{:0>2}", datetime.day())?;
+        //                   ^-- delimiting hyphen and 0-padded, right aligned, 2-digit day
+        if value.precision == Precision::Day {
+            write!(self.output, "T")?;
+            return Ok(());
+        }
+
+        write!(
+            self.output,
+            "T{:0>2}:{:0>2}",
+            datetime.hour(),
+            datetime.minute()
+        )?;
+        //                   ^-- delimiting T, formatted hour, delimiting colon, formatted minute
```

Comment:
`cargo fmt` has thrown off the spacing for this comment. Could you move it up a couple of lines and line up the `^--` to the `T`?